src/App.tsx

```diff
@@ -1,16 +1,27 @@
 import './App.scss';
+import { useState } from 'react';
 import { MoviesList } from './components/MoviesList';
 import { NewMovie } from './components/NewMovie';
 import moviesFromServer from './api/movies.json';
+import { Movie } from './types/Movie';
 
 export const App = () => {
+  const [movies, setMovies] = useState(moviesFromServer);
+
+  const addMovie = (movie: Movie) => {
+    setMovies((prev) => [...prev, movie]);
+  };
+
   return (
     <div className="page">
       <div className="page-content">
-        <MoviesList movies={moviesFromServer} />
+        <MoviesList movies={movies} />
       </div>
       <div className="sidebar">
-        <NewMovie /* onAdd={(movie) => {}} */ />
+        <NewMovie onAdd={(movie: Movie) => {
+          addMovie(movie);
+        }}
+        />
```

Comment:
```suggestion
        <NewMovie onAdd={addMovie} />
```
`addMovie` is a function that accepts `movie` argument. No need to add additional function wrapper, just use reference to `addMovie` function

FYI only